doc/reference/kernel/smp/smp.rst

```diff
@@ -297,3 +297,8 @@ Note that while SMP requires :kconfig:`CONFIG_USE_SWITCH`, the reverse is not
 true.  A uniprocessor architecture built with :kconfig:`CONFIG_SMP` set to No might
 still decide to implement its context switching using
 :c:func:`arch_switch`.
+
+API Reference
+**************
+
+.. doxygengroup:: spinlock_apis
```

Comment:
is this really the page where spinlock API should be placed?

Reply:
Please feel free to suggest a better location 😁

Reply:
@nashif @andyross any suggestions?

Reply:
I am fine with that.

Reply:
As originally suggested on Discord, I'd prefer a spinlock page under synchronisation as well, but let's get this in. We can move it around in a separate PR.